utils/cli/functions.py

```diff
@@ -61,6 +61,64 @@ def update_config(
     return updated_config
 
 
+def update_config_bulk(
+    config: dict | None = None,
+    keys: list | None = None,
+    console: Console | None = None,
+    updated_config: dict | None = None,
+) -> dict:
+    """Function that interactively and recursively updates a configuration,
+    based on user input. This is slightly different from the update_config, which
+    allows users to update any field in one task. This function keeps track of
+    requested updates to all tasks and applies them downstream.
+    Args:
+        config (dict): Configuration object to update.
+        keys (list): List of keys to update.
+        console (Console): Rich console object for printing.
+        updated_config (dict): Updated configuration object.
+    Returns:
+        dict: Updated configuration object.
+    """
+
+    # Initialize updated_config if this is the top-level call
+    if updated_config is None:
+        updated_config = {}
+
+    for key in keys:
+        to_modify = Confirm.ask(
+            f":key: Would you like to modify [bold green]{key}[/bold green]?"
+        )
+
+        if to_modify:
+            val_to_modify = config.get(key)
+            console.print("[bold] :pencil: Modifying:\n")
+            console.print(val_to_modify)
+
+            if isinstance(val_to_modify, dict):
+                # Handle nested dictionary
+                updated_config[key] = {}
+                update_config(
```

Comment:
```suggestion
                updated_config = update_config(
```
Do we need to save it to a variable? Or does it modify in place?

Reply:
Modifies in place. Ideally we'd have a solution without recursion + mutating objects, but I'm not sure how to do that with arbitrary nested dictionaries/arrays.